doc/holochain_101/src/SUMMARY.md

```diff
@@ -4,22 +4,47 @@
   - [Overview](./overview.md)
   - [How to contribute](./how_to_contribute.md)
 - [Planning a dApp](./planning_a_dapp.md)
-- [Building Holochain Apps](./building_apps.md)
+- [Building Holochain Apps: Intro](./building_apps.md)
+  - [Building for Different Platforms](./building_for_different_platforms.md)
+    - [Building For Android](./building_for_android.md)
   - [Intro to DNA](./intro_to_dna.md)
   - [Intro to Command Line Tools](./intro_to_command_line_tools.md)
   - [Create A New Project](./new_project.md)
-  - [Using Zome API global variables](./zome/api_globals.md)
-  - [Using Zome API functions](./zome/api_functions.md)
-  - [Using callback functions](./zome/callback_functions.md)
-  - [Lifecycle of an Entry](./lifecycle_of_an_entry.md)
-  - [Links Entries](./links_entries.md)
-  - [WebAssembly](./webassembly.md)
+  - [Configuring an App](./configuring_an_app.md)
+  - [Updating to Alpha 2](./alpha_migrate.md)
+- [Building Holochain Apps: Zome Code](./zome/welcome.md)
+  - [Adding a Zome](./zome/adding_a_zome.md)
+  - [Capabilities](./zome/capabilities.md)
+  - [Entry Definitions](./zome/entry_definitions.md)
+  - [Read & Write Data Operations](./zome/read_and_write.md)
+  - [Entry Validation](./zome/entry_validation.md)
+  - [Genesis](./zome/genesis.md)
+  - [Node to Node Messaging](./zome/node_to_node_messages.md)
+  - [Cryptographic Signatures](./zome/signatures.md)
+  - [Calling Other Zomes](./zome/calling_other_zomes.md)
+  - [Bundling](./zome/bundling.md)
+  - [Linking](./zome/linking.md)
+  - [Emitting Signals](./zome/emitting_signals.md)
+  - [Compiling to WebAssembly](./zome/compiling_to_webassembly.md)
+  - [Writing in Rust](./zome/rust.md)
+  - [Writing in Assemblyscript](./zome/assemblyscript.md)
+  - [Understanding WebAssembly](./zome/understanding_webassembly.md)
```

Comment:
This is the kind of feedback I'm loooking for. :)

This feels tied in with an iterated change I was considering, of moving "Writing in Rust" and "Writing in WebAssembly" up into the end of the "Building: Intro" section. 
I am actually not even sure what I'd put in Understanding WebAssembly yet... that would differ from Compiling to WebAssembly. 

Actually, I was thinking about contextualizing how it actually runs WebAssembly, since if you were hardcore, you could just write WebAssembly code... or maybe think about writing an HDK either

Reply:
Given that this sounds like it'll be an ongoing conversation because the structure wants to gel some more, I'm gonna click 'Resolve'.

For most developers I think it'd be sufficient to explain why we chose WebASM, how the WebASM code in the DNA interfaces with Holochain (that is, explain that Holochain calls your WebASM in response to ____ sorts of actions) and point people to other resources to explain what WebASM is actually for.

The 'how Holochain interfaces with WebASM' seems like it might be worth fleshing out, because some app developers might not understand the difference between callbacks, zome functions, and API built-ins.